doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,724 @@
+launch our application's window.
+
+This basically leaves us with two bad scenarios. Either we're a Console
+application, and a console window always flashes on screen for every
+non-commandline invocation of the Terminal, or we're a Windows application, and
+console output we log (including help messages) can get mixed with shell output.
+Neither of these are particularly good.
+
+`python` et. al. often ship with _two_ executables, a `python.exe` which is a
+Console application, and a `pythonw.exe`, which is a Windows application. This
+however has led to [loads of confusion](https://stackoverflow.com/a/30313091),
+and even with plentiful documentation, would likely result in users being
+confused about what does what. For situations like launching the Terminal in the
+CWD of `explorer.exe`, users would need to use `wtw.exe -d .` to prevent the
+console window from appearing. However, when calling Windows Terminal from a
+commandline environment, users who call `wtw.exe /?` would likely get unexpected
+behavior, because they should have instead called `wt.exe /?`.
+
+To avoid this confusion, I propose we follow the example of `msiexec /?`. This
+is a Windows application that uses a `MessageBox` to display its help text.
+While this is less convenient for users coming exclusively from a commandline
+environment, it's also the least bad option available to us.
+* It's less confusing than having control returned to the shell
+* It's not as bad as forcing the creation of a console window for
+  non-commandline launches.
+* There's precedent for this kind of dialog (we're not inventing a new pattern
+  here.)
+
+### What happens if `new-tab` isn't the first command?
+
+**TODO**: What should we do if we encounter the following?
+
+```sh
+wt.exe split-pane -v ; new-tab
+```
+
+In the future, maybe we could presume in this case that the commands are
+intended for the current Windows Terminal window, though that's not
+functionality that will arrive in 1.0. Even when sessions are supported like
+that, I'm not sure that when we're parsing a commandline, we'll be possible to
+know what session we're currently running in.That might make it challenging to
+dispatch this kind of command to "the current WT window".
+
+Additionally, what would happen if this was run in a `conhost` window, that
+wasn't attached to a Terminal session? We wouldn't be able to tell _the current
+session_ to `split-pane`, since there wouldn't be one. What would we do then?
+Display an error message somehow?
+
+I don't believe that implying the _current Windows Terminal session_ is the
+correct behavior here. Instead we should either:
+* Assume that there's an implicit `new-tab` command that's run first, to create
+  the window, _then_ run `split-pane` in that tab.
+* Immediately display an error that the commandline is invalid, and that a
+  commandline should start with a `new-tab ; `?
+
+**TODO**: This ^ is left for discussion. In my initial implementation, I

```

Comment:
I don't really care for implicits. I think if you specify something we cannot resolve, we should say that it's not valid.

Reply:
Yea I was on the fence about this anyways. This also makes it a little easier to handle the paragraph below "don't open a terminal tab just to open the settings".

As far as error text, something like 
```
"Invalid wt.exe commandline provided. A {{command}} must be preceded by a new-tab command" 
```
sound good?

Reply:
Resolving per the discussion on the broader PR.